### PR TITLE
LPS-150749

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://issues.liferay.com/browse/LPS-150749

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

File: modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/FileEntryFileEntryItemSelectorReturnTypeResolver.java [hunk 1]
```diff
@@ -93,6 +93,8 @@ public String getValue(FileEntry fileEntry, ThemeDisplay themeDisplay)
 
 				return null;
 			}
+		).put(
+			"size", fileEntry.getSize()
 		).put(
 			"title", fileEntry.getTitle()
 		).put(
```
Comment (carolmariaabb) [inline modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/FileEntryFileEntryItemSelectorReturnTypeResolver.java:100]: @adolfopa can you take a look at this change? Thanks!

**Comment by liferay-forms** [comment]
ci:test:relevant

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Many thanks for your work, @carolmariaabb. Let's not create this service for now. Since this entry validations will most likely to be used on other business types, let's think of a more generic approach later. Feel free to just copy the logic in the ObjectEntryLocalService and we'll refactor it later.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -1611,6 +1612,53 @@ private void _updateTable(
 		}
 	}
 
+	private void _validateFileExtension(
```
Comment (rodrigopaulino) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1615]: Could you test if this works for this validation? Let me know your opinion if you think it's worth changing.
```
ObjectFieldSetting objectFieldSetting =
	_objectFieldSettingPersistence.fetchByOFI_N(
		objectFieldId, "acceptedFileExtensions");

String objectFieldSettingValue = objectFieldSetting.getValue();

String[] acceptedFileExtensions = objectFieldSettingValue.split(
	"\\s*,\\s*");

if (!ArrayUtil.contains(acceptedFileExtensions, fileExtension)) {
	throw new ObjectEntryValuesException.InvalidFileExtension(
		fileExtension, objectFieldName);
}
```

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 3]
```diff
@@ -1728,9 +1776,33 @@ private void _validateValues(
 			return;
 		}
 
-		String dbType = objectField.getDBType();
+		if (StringUtil.equals(
```
Comment (rodrigopaulino) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1779]: I was able to save one tab by:
```
DLFileEntry dlFileEntry = _dlFileEntryLocalService.fetchDLFileEntry(
	GetterUtil.getLong(entry.getValue()));

if (dlFileEntry != null) {
	_validateFileExtension(
		dlFileEntry.getExtension(), objectField.getObjectFieldId(),
		objectField.getName());
	_validateFileSize(
		dlFileEntry.getSize(), objectField.getObjectFieldId(),
		objectField.getName());

	return;
}

if (objectField.isRequired()) {
	throw new ObjectEntryValuesException.Required(
		objectField.getName());
}
```
Do you think it's worth changing?

**Review by rodrigopaulino — COMMENTED** [review]
Hey, @carolmariaabb. I've made some suggestions below. Nothing very necessary, but let me know what you think.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 3 above)
Comment (carolmariaabb) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1770]: I applied the change, thanks!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1615]: I applied the change, thanks!

**Review by miltonmc — APPROVED** [review]
LGTM!

**Comment by miltonmc** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @carolmariaabb, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @carolmariaabb, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:stop

**Comment by liferay-forms** [comment]
Hey @carolmariaabb, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**